pkg/network/common/egressip.go

```diff
@@ -290,6 +300,174 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *networkv1.HostSubnet) {
 	eit.syncEgressIPs()
 }
 
+func nodeIsReady(node *v1.Node) bool {
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady {
+			if cond.Status == v1.ConditionFalse || cond.Status == v1.ConditionUnknown {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// SetMonitorNodes updates EgressIPTracker with the node data provided and
+// starts a go-routine (if one is not already running) which polls monitorNodes
+// to check if they are ready and reachable.
+func (eit *EgressIPTracker) SetMonitorNodes(monitorNodes []*NodeEgress) {
```

Comment:
So this fixes the problem of "not overwriting the tracker's data with blank objects from the caller" now, but it makes the whole thing kind of weird... We mostly aren't actually using the objects passed in from the caller, so we didn't really need to export NodeEgress as a type; it would have made more sense to just use a different type here that's just `struct { Name string; IP string }`, or just use a `map[string]string`...